### PR TITLE
[Table] When writing code, recompile only the type of file that was changed

### LINKED ISSUE #1441: Make code compilation faster when dev'ing on Table
When I press `Cmd + S` in a `Table` file with `npm run watch` running, all of the following execute regardless of the type of file I was just editing:

```tsx
[11:44:43] Working directory changed to ~/Developer/blueprint-external
[11:44:46] Using gulpfile ~/Developer/blueprint-external/gulpfile.js
[11:44:46] Starting 'icons'...
[11:44:46] Finished 'icons' after 36 ms
[11:44:46] Starting 'tsc-core'...
[11:44:46] Starting 'sass-variables'...
[11:44:46] Finished 'sass-variables' after 274 ms
[11:44:46] Starting 'sass-core'...
[11:44:46] core: compiling 74 typescript files
[11:44:49] Finished 'tsc-core' after 3.53 s
[11:44:49] Starting 'tsc-table'...
[11:44:49] table: compiling 43 typescript files
[11:44:51] Finished 'tsc-table' after 1.85 s
[11:44:53] write blueprint.css.map
[11:44:53] write blueprint.css
[11:44:53] Finished 'sass-core' after 7.1 s
[11:44:53] Starting 'sass-table'...
[11:44:54] write table.css.map
[11:44:54] write table.css
[11:44:54] Finished 'sass-table' after 302 ms
```

Would be a huge productivity win to speed this up.

### PR DESCRIPTION
**Pull request by cmslewis**
#### Changes proposed in this pull request:

*Before:*
- Make a code change to any file in `packages/table`, and SCSS/TS/all code is recompiled, taking forever (~20sec per file save).

*After:*
- Make a code change to a SCSS file in `packages/table` (e.g.), and only SCSS code is recompiled, saving time (4 sec).
    - Same for TS code (8 sec).
    - Same for `Table` `preview/` code (8 sec). 

#### Screenshot

Note that when file-type specific tasks are triggered, only the build tasks relevant to that file type are triggered as a consequence.

```tsx
> @blueprintjs/table@1.21.0 build:preview /Users/clewis/Developer/blueprint-external/packages/table
> webpack --config config/webpack.config.preview.js

ts-loader: Using typescript@2.2.1 and /Users/clewis/Developer/blueprint-external/packages/table/preview/tsconfig.json
Hash: c862b3d5710101fa2f83
Version: webpack 1.13.3
Time: 8252ms
              Asset     Size  Chunks             Chunk Names
 fonts/icons-16.eot  80.5 kB          [emitted]
fonts/icons-16.woff  80.3 kB          [emitted]
 fonts/icons-16.ttf  80.3 kB          [emitted]
 fonts/icons-20.eot  82.4 kB          [emitted]
fonts/icons-20.woff  82.3 kB          [emitted]
 fonts/icons-20.ttf  82.2 kB          [emitted]
 features.bundle.js  2.35 MB       0  [emitted]  features
    index.bundle.js  2.35 MB       1  [emitted]  index
       features.css   294 kB       0  [emitted]  features
          index.css   294 kB       1  [emitted]  index
 [183] ../core/dist/index.js 618 bytes {0} {1} [built]
 [184] ../core/dist/accessibility/index.js 556 bytes {0} {1} [built]
 [185] ../core/dist/accessibility/focusStyleManager.js 1.26 kB {0} {1} [built]
 [186] ../core/dist/common/interactionMode.js 2.44 kB {0} {1} [built]
 [187] ../core/dist/common/index.js 1.16 kB {0} {1} [built]
 [188] ../core/dist/common/abstractComponent.js 3.04 kB {0} {1} [built]
 [190] ../core/dist/common/utils.js 5.2 kB {0} {1} [built]
 [191] ../core/dist/common/errors.js 4.58 kB {0} {1} [built]
 [192] ../core/dist/common/colors.js 2.51 kB {0} {1} [built]
 [193] ../core/dist/common/intent.js 768 bytes {0} {1} [built]
 [194] ../core/dist/common/position.js 1.85 kB {0} {1} [built]
 [195] ../core/dist/common/props.js 1.85 kB {0} {1} [built]
 [196] ../core/dist/common/tetherUtils.js 3.91 kB {0} {1} [built]
 [197] ../core/dist/common/classes.js 7.07 kB {0} {1} [built]
 [198] ../core/dist/common/keys.js 657 bytes {0} {1} [built]
 [199] ../core/dist/generated/iconClasses.js 15.9 kB {0} {1} [built]
 [200] ../core/dist/generated/iconStrings.js 11 kB {0} {1} [built]
 [392] ../core/dist/compatibility/index.js 546 bytes {0} {1} [built]
 [393] ../core/dist/compatibility/browser.js 895 bytes {0} {1} [built]
    + 470 hidden modules
Child extract-text-webpack-plugin:
        + 8 hidden modules
Child extract-text-webpack-plugin:
        + 2 hidden modules

> @blueprintjs/table@1.21.0 build:sass /Users/clewis/Developer/blueprint-external/packages/table
> $(cd ..; npm bin)/gulp sass-table

[12:44:24] Working directory changed to ~/Developer/blueprint-external
[12:44:27] Using gulpfile ~/Developer/blueprint-external/gulpfile.js
[12:44:27] Starting 'icons'...
[12:44:27] Finished 'icons' after 52 ms
[12:44:27] Starting 'sass-variables'...
[12:44:27] Finished 'sass-variables' after 297 ms
[12:44:27] Starting 'sass-core'...
[12:44:30] write blueprint.css.map
[12:44:30] write blueprint.css
[12:44:30] Finished 'sass-core' after 2.97 s
[12:44:30] Starting 'sass-table'...
[12:44:30] write table.css.map
[12:44:30] write table.css
[12:44:30] Finished 'sass-table' after 437 ms

> @blueprintjs/table@1.21.0 build:typescript /Users/clewis/Developer/blueprint-external/packages/table
> $(cd ..; npm bin)/gulp tsc-table

[12:44:34] Working directory changed to ~/Developer/blueprint-external
[12:44:38] Using gulpfile ~/Developer/blueprint-external/gulpfile.js
[12:44:38] Starting 'icons'...
[12:44:38] Finished 'icons' after 50 ms
[12:44:38] Starting 'tsc-core'...
[12:44:39] core: compiling 74 typescript files
[12:44:43] Finished 'tsc-core' after 4.61 s
[12:44:43] Starting 'tsc-table'...
[12:44:43] table: compiling 45 typescript files
[12:44:46] Finished 'tsc-table' after 2.9 s
```

### CODE REVIEW

**Comment by giladgray** [comment]
@cmslewis pro tip: in terminal config file (maybe .zshrc), add `alias gulp=$(npm bin)/gulp`

File: packages/table/package.json [hunk 1]
```diff
@@ -8,6 +8,8 @@
   "unpkg": "dist/table.bundle.js",
   "scripts": {
     "build:gulp": "$(cd ..; npm bin)/gulp tsc-table sass-table",
```
Comment (giladgray) [inline packages/table/package.json]: could replace this task with the two separates you defined below

**Comment by cmslewis** [comment]
@giladgray thanks. my `gulp` foo is weak.

File: packages/table/package.json (hunk 1 above)
Comment (cmslewis) [inline packages/table/package.json]: Yeah, wasn't sure if running in parallel would be safe, so wanted to get your thoughts before playing with that.

Comment (giladgray) [inline packages/table/package.json]: if i understand your concern properly: gulp always runs with maximum parallelization (based on task deps) so we're already doing that

**Comment by cmslewis** [comment]
Actually this change is incorrect: `build:preview` still needs to run regardless, which really doesn't save much time. @giladgray if you have time, could I consult with you on changing the rebuild process for `packages/table` code? Generally seems so much slower than it could be, so would love to know your thoughts given your experience with build stuff.

**Pull request closed without merging**